onnxscript/tests/function_libs/torch_lib/ops_test.py

```diff
@@ -153,11 +157,12 @@ def run_test_output_match(
             ),
             kwargs=repr(cpu_sample.kwargs),
         ):
-            skip_reason = _should_skip_test_sample(op.name, cpu_sample)
-            if skip_reason is not None:
-                # Cannot use self.skip because pytest would skip the entire test
-                warnings.warn(f"skipped sample {i}. Reason: {skip_reason}", stacklevel=1)
-                continue
+            skip_xfail_or_not: Optional[Tuple[str, Callable]] = _should_skip_xfail_test_sample(
+                op.name, cpu_sample
+            )
+            if skip_xfail_or_not is not None:
+                skip_xfail_or_not[1](skip_xfail_or_not[0])
```

Comment:
How is xfail asserted? I imaging we actually catch the error before raising xfail